### ISSUE TITLE
Xbox compatibility 

### ISSUE THREAD

**Issue by BrianAutry** [vladimirprasanna/prw#1]
I tested the firmware on Xbox... It did not work. The G Pro Xbox/PC wheelbase only gives me the option to run it in PC mode unless the original Logitech Xbox wheel is attached. I am going to try to test it in PC mode on the Xbox using a collective minds DriveHub adapter. 
Thank you!

**Comment by TheTrainGoes** [comment]
@BrianAutry and @vladimirprasanna my understanding is as follows for the three Logitech G Pro wheel bases:

Logitech G Pro Playstation - Has the authentication chip inside the base
Logitech G Pro PC - No authentication needed
Logitech G Pro Xbox - Is just the Logitech G Pro PC, authentication is done by the wheel rims

This allows you to have the Logitech G Pro Playstation and an Xbox rim and use them on either system.

For this code to work on an Xbox you will need to have some sort of authentication.

You can check out a few example projects that allow your passthrough authentication for Xbox like GP2040-CE and Santroller.

**Comment by BrianAutry** [comment]
> @BrianAutry and @vladimirprasanna my understanding is as follows for the three Logitech G Pro wheel bases:
> 
> Logitech G Pro Playstation - Has the authentication chip inside the base Logitech G Pro PC - No authentication needed Logitech G Pro Xbox - Is just the Logitech G Pro PC, authentication is done by the wheel rims
> 
> This allows you to have the Logitech G Pro Playstation and an Xbox rim and use them on either system.
> 
> For this code to work on an Xbox you will need to have some sort of authentication.
> 
> You can check out a few example projects that allow your passthrough authentication for Xbox like GP2040-CE and Santroller.
@TheTrainGoes 
yes, you're correct... the authentication chip is in the wheel for Xbox. 

it should still work with the Collective Minds Drive Hub converter... the problem right now is that since Logitech updated their firmware for the new RS Hub the Drive Hub is not working. I have contacted Collective Minds and they know they have a problem with the Drive Hub not working with the G pro.... As soon as the Drive Hub firmware gets updated to work with the G Pro again the Vota wheel should work in PC mode with the Drive Hub on Xbox.

**Comment by BrianAutry** [comment]
@vladimirprasanna @TheTrainGoes    I was able to get it to work on Xbox with the Collective Minds Drive Hub adapter. Only on games without TrueForce right now. The Drive Hub firmware still needs to be updated before it will work on games with TrueForce. Drive Hub support knows about the problem and they are working on it.

**Comment by TheTrainGoes** [comment]
@BrianAutry - If you start the base with a connected xbox wheel and choose xbox then wait for connection, can you then remove the wheel and move to a generic one?  

Xbox only requires auth at the time of initial connection.  Curious to know if this is also the case with these bases or if there is some sort of reset that happens in the base when you change rims.

**Comment by BrianAutry** [comment]
@TheTrainGoes --- the wheel base shuts off as soon as you remove the wheel....

**Comment by vladimirprasanna** [comment]
From my tests, the compatibility with xbox is only partial. I managed to make the prw recognized by xbox for only 10 seconds, unfortunately there is an encrypted communication between the base and the wheel and it changes every time

**Comment by BrianAutry** [comment]
I did get it to work with the Drive Hub converter... but only one game so far. It works on the original Assetto Corsa. The Drive Hub firmware has not been updated yet... after collective minds updates the Drive Hub firmware it should work with any game.

**Comment by TheTrainGoes** [comment]
Has anyone done a disassembly of the two hubs?  

As soon as my Xbox ones ship I can take one apart but I won't be able to compare it to anything.  

Presumably a dongle device like the MagicBoots Xbox could be used if the necessary code was added into the firmware.

You can see how we implemented Xbox in GP2040-CE here:
https://github.com/OpenStickCommunity/GP2040-CE/tree/main/headers/drivers/xbone

Xbox only needs to do an initial auth call.  Once successful there are no additional calls while the device is connected.
